include/nbl/video/IGPUCommandPool.h

```diff
@@ -31,13 +66,219 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         // Vulkan: const VkCommandPool*
         virtual const void* getNativeHandle() const = 0;
 
+        core::CMemoryPool<core::GeneralpurposeAddressAllocator<uint32_t>, core::default_aligned_allocator, false, uint32_t> m_commandSegmentPool;
+
     protected:
         virtual ~IGPUCommandPool() = default;
 
         core::bitflag<E_CREATE_FLAGS> m_flags;
         uint32_t m_familyIx;
 };
 
+class alignas(COMMAND_ALIGNMENT) IGPUCommandPool::ICommand
+{
+public:
+    virtual ~ICommand() {}
+
+    // static void* operator new(std::size_t size) = delete;
+    static void* operator new[](std::size_t size) = delete;
+    // static void* operator new(std::size_t size, std::align_val_t al) = delete;
+    static void* operator new[](std::size_t size, std::align_val_t al) = delete;
+
+    // static void operator delete  (void* ptr) = delete;
+    static void operator delete[](void* ptr) = delete;
+    static void operator delete  (void* ptr, std::align_val_t al) = delete;
+    static void operator delete[](void* ptr, std::align_val_t al) = delete;
+    static void operator delete  (void* ptr, std::size_t sz) = delete;
+    static void operator delete[](void* ptr, std::size_t sz) = delete;
+    static void operator delete  (void* ptr, std::size_t sz, std::align_val_t al) = delete;
+    static void operator delete[](void* ptr, std::size_t sz, std::align_val_t al) = delete;
+
+    uint32_t m_size;
+protected:
+    ICommand(uint32_t size) : m_size(size) {}
+};
+
+class alignas(COMMAND_SEGMENT_ALIGNMENT) IGPUCommandPool::CommandSegment
+{
+public:
+    struct Iterator;
+
+    struct params_t
```

Comment:
not sure it should be called `params_t` cause its not being passed as a parameter to anything... maybe `header_t` is better

Reply:
also why are all the members public and dangling?

Reply:
Made them `private`.
Resolved.